src/en/config-azure.md

```diff
@@ -1,123 +1,223 @@
+
+
+# Prerequisites and installation of Juju and the Azure CLI tool
+
+ - An Azure account is required. See http://azure.microsoft.com.
+
+ - The Juju devel PPA (may change) is needed.
+
+ - The Azure CLI tool is used to both gather information and to perform
+   necessary actions.
+
+ - The Juju client (the host running the below commands) will need the ability
+   to contact the Azure infrastructure on TCP ports 22 and 17070.
+
+Proceed to install the software.
 
 ```bash
-juju generate-config
+sudo apt-add-repository -y ppa:juju/devel
+sudo apt-get update
+sudo apt-get install -y juju-core nodejs-legacy npm
+sudo npm install -g azure-cli
+```
+
+## Azure CLI tool preliminaries
+
+The Azure CLI tool gets installed here:
+
+```bash
+ls -lh /usr/bin/azure
+lrwxrwxrwx 1 root root 39 Jan 18 22:58 /usr/bin/azure -> ../lib/node_modules/azure-cli/bin/azure
+```
+
+Confirm it's installed correctly by viewing its online help. Then put it in
+*Azure Resource Manager* mode and log in:
+
+```bash
+azure help
+azure config mode arm
+azure login
 ```
 
-This will generate a file, `environments.yaml`, which will live in your
-`~/.juju/` directory (and will create the directory if it doesn't already
-exist).
+You will be prompted to visit a website to enter the provided code. It will
+therefore be easier to perform this on a (graphical) Ubuntu desktop.
+
+
+# Configuring for Microsoft Azure
 
-**Note:** The above command will not overwrite your existing environments.yaml
-file, or output to stdout. In order to see the boilerplate environments.yaml on
-stdout you need to append the `--show` option. This is helpful if you have an
-existing environments.yaml and just need to add a section. For example:
+If this is a new Juju install then you do not yet have a
+`~/.juju/environments.yaml` file. Create one with
+
+```bash
+juju generate-config
+```
+
+If it does exist (but it was created with an older version of Juju), first move
+it out of the way (back it up) and *then* generate a new one. Alternatively,
+you can output a generic file to screen (STDOUT) and paste the Azure parts into
+your existing file:
 
 ```bash
 juju generate-config --show
 ```
 
-You can then copy and paste the needed section.
+The file will contain a section for the Azure provider.
+
+
+## Configure and bootstrap
+
+Values will need to be found for the following parameters:
+
+ - storage-account-type
+ - location
+ - subscription-id
+ - application-password
+ - application-id
+ - tenant-id
+
+#### `storage-account-type`
+
+Use the generic value of 'Standard_LRS' for this.
+
+### `location`
+
+Choose an [Azure region](https://azure.microsoft.com/en-us/regions/). Here we will use
+'East US'.
+
+### `subscription-id`
+
+List your account and get the subscription ID, the **SUB_ID**:
 
-The generic configuration sections generated for Windows Azure will look
-something like this:
+```bash
+azure account list
+```
+
+Set its value in the terminal. It will look similar to:
```

Comment:
I would change this a bit to make sure people understand that they are fetching a value from above, and setting an environment variable to be equal to that (e.g. Mac  users may not understand what these things are).

Something like, "This value can now be stored in a local environment variable, by entering:

xxxxxxxxx